Fixes #34555 - redesign scheduling step job wizard

MariaAga opened the pull request:
![Screenshot from 2022-03-29 09-37-26](https://user-images.githubusercontent.com/30431079/160559009-169eedf5-4fc5-4eaa-94d7-6e3e3e5fdc59.png)
![Screenshot from 2022-03-29 09-37-33](https://user-images.githubusercontent.com/30431079/160559010-0be94d6a-74b3-4d4d-a2da-eeba6719c2a1.png)
![Screenshot from 2022-03-29 09-38-57](https://user-images.githubusercontent.com/30431079/160559108-43ca8faf-6989-4175-b11a-84eeabf05590.png)

MariaAga commented on package.json:
```diff
@@ -21,11 +21,11 @@
-    "@theforeman/stories": "^8.16.0",
-    "@theforeman/test": "^8.16.0",
-    "@theforeman/vendor-dev": "^8.16.0",
+    "@theforeman/builder": "^10.1.0",
```
The core pr will be probably merged before this one https://github.com/theforeman/foreman/pull/9110

adamruzicka replied:
The core PR was indeed merged before this one, resolving.

MariaAga commented:
Missing: clearing function for the `Future execution` step datepickers ( probably buttons)

adamruzicka commented:
Looks nice, the only thing I was surprised with was that you can select future or recurring execution and then submit the job without filling in any details. Not saying its wrong, I just didn't see it coming

MariaAga commented:
Not sure what a solution to the surprise you mentioned, but uses can see the pre-selected values at the next step and also in the review step
Re did this pr a bit, main changes are:
* Added a few visual dividers 
* moved the query type to the `type of execution` step
* Added clear buttons for the future execution step
* Made starts at/before mandatory
* Pre selected starts at to be now

MariaAga commented:
@Ron-Lavi can you take a look?

adamruzicka commented:
@MariaAga needs a rebase

@Ron-Lavi I'm ok with merging the smaller PRs on my own, but I'd really appreciate your input on this one

jeremylenz commented:
@adamruzicka @MariaAga taking a look, stand by for review :)

jeremylenz commented on webpack/JobWizard/steps/Schedule/ScheduleType.js:
```diff
@@ -1,28 +1,96 @@
+          name="schedule-type-reccuring"
+          id="schedule-type-reccuring"
+          label={__('Recurring execution')}
+          body={__('Set up a reccuring execution for a job.')}
```
```suggestion
          body={__('Execute the job on a repeating schedule.')}
```

jeremylenz commented on webpack/JobWizard/JobWizard.js:
```diff
@@ -187,18 +191,80 @@ export const JobWizard = () => {
+              },
+            ]
+          : []),
+        ...(scheduleValue.scheduleType === SCHEDULE_TYPES.RECCURING
```
```suggestion
        ...(scheduleValue.scheduleType === SCHEDULE_TYPES.RECURRING
```

Recurring has two R's, not two C's.. should be changed everywhere ;)

jeremylenz commented on webpack/JobWizard/steps/Schedule/ScheduleReccuring.js:
```diff
@@ -0,0 +1,275 @@
+import React, { useEffect, useState } from 'react';
```
file name spelling should be corrected `ScheduleRecurring.js`

jeremylenz commented on webpack/JobWizard/steps/Schedule/ScheduleFuture.js:
```diff
@@ -0,0 +1,125 @@
+      setValid(true);
+      setError(null);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
See previous comment about disabling this rule.

jeremylenz reviewed (commented):
Thanks @MariaAga! Seems to work well; some comments below.

Not sure if this is related, but when I select Recurring and try to submit, I'm getting a console error:

```
Uncaught TypeError: repeatData.at.split is not a function
    at getCronLine (submit.js:39:23)
    at submit (submit.js:76:26)
    at onSave (JobWizard.js:303:9)
    at Wizard.onNext (Wizard.js:34:1)
    at HTMLUnknownElement.callCallback (react-dom.development.js:188:1)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:237:1)
    at invokeGuardedCallback (react-dom.development.js:292:1)
    at invokeGuardedCallbackAndCatchFirstError (react-dom.development.js:306:1)
    at executeDispatch (react-dom.development.js:389:1)
    at executeDispatchesInOrder (react-dom.development.js:414:1)
```

MariaAga commented:
Thanks @jeremylenz , added a commit to address your comments

jeremylenz reviewed (approved):
Changes look good, thanks! (also Ruby test failures seem unrelated, they're about the template name change thing)

adamruzicka commented:
Thank you @MariaAga & @jeremylenz !

jeremylenz commented on webpack/JobWizard/steps/Schedule/ScheduleReccuring.js:
```diff
@@ -0,0 +1,275 @@
+    } else {
+      setValid(false);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
@MariaAga This article is what I was trying to explain.. Dan explains it much better here

https://beta.reactjs.org/learn/you-might-not-need-an-effect
